File: clients/algoliasearch-client-javascript/packages/client-common/src/types/chunkedHelperOptions.ts
/**
 * Shared configuration for chunked helpers that poll for task completion.
 * Designed to grow over time so future helper configs can be added without ballooning option types.
 */
export type ChunkedHelperOptions = {
  /**
   * The maximum number of retries when polling for task completion. 100 by default.
   */
  maxRetries?: number | undefined;
};

File: clients/algoliasearch-client-dart/packages/client_core/lib/src/config/chunked_helper_options.dart
/// Default maximum number of retries for chunked helpers and wait operations.
const int defaultMaxRetries = 100;

/// Default maximum number of retries used by `replaceAllObjects`.
const int defaultReplaceAllObjectsMaxRetries = 800;

/// Optional configuration for chunked helpers that batch records and poll for task completion.
///
/// Designed to grow over time; future shared helper config (e.g. timeout) can be
/// added here without widening every helper's parameter list.
final class ChunkedHelperOptions {
  /// Maximum number of retries when polling for task completion.
  /// Defaults to [defaultMaxRetries].
  final int maxRetries;

  ChunkedHelperOptions({int? maxRetries})
      : maxRetries = maxRetries ?? defaultMaxRetries {
    if (this.maxRetries < 1) {
      throw ArgumentError('`maxRetries` must be greater than 0');
    }
  }
}
